util/promlint/promlint.go

```diff
@@ -205,6 +210,64 @@ func lintHistogramSummaryReserved(mf dto.MetricFamily) []Problem {
 	return problems
 }
 
+// lintMetricTypeInName detects when metric types are included in the metric name.
+func lintMetricTypeInName(mf dto.MetricFamily) []Problem {
+	var problems problems
+	n := strings.ToLower(mf.GetName())
+
+	for i, t := range dto.MetricType_name {
+		if i == int32(dto.MetricType_UNTYPED) {
+			continue
+		}
+
+		typename := strings.ToLower(t)
+		if strings.Contains(n, "_"+typename+"_") || strings.HasSuffix(n, "_"+typename) {
+			problems.Add(mf, fmt.Sprintf(`metric name should not include type '%s'`, typename))
+		}
+	}
+	return problems
+}
+
+// lintReservedChars detects colons in metric names.
+func lintReservedChars(mf dto.MetricFamily) []Problem {
+	var problems problems
+	if strings.Contains(mf.GetName(), ":") {
+		problems.Add(mf, "metric names should not contain ':'")
+	}
+	return problems
+}
+
+var camelCase *regexp.Regexp = regexp.MustCompile(`[a-z][A-Z]`)
```

Comment:
```suggestion
var camelCase = regexp.MustCompile(`[a-z][A-Z]`)
```

Reply:
Thanks @mdlayher . Also, that was FAST response :+1:  ha ha